app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -140,9 +140,20 @@ def link_identity_from_session_data
       link_identity(
         ial: resolved_authn_context_int_ial,
         rails_session_id: session.id,
+        email_address_id: email_address_id,
       )
   end
 
+  def email_address_id
+    return nil unless IdentityConfig.store.feature_select_email_to_share_enabled
+    return session[:sp_email_id] if session[:sp_email_id].present?
+    sp = sp_session['issuer']
+    identity = current_user.identities.where(service_provider: sp)
+    email_id = identity.pluck('email_address_id')[0]
```

Comment:
Can there be multiple identities for the user for a given issuer?

- If yes, how do we know we're picking the right one when we grab the first email?
- If no, then should we just find the singular identity record, rather than filtering as an array?

Reply:
no, we have a unique index on `[user_id, service_provider]` in that table

Reply:
If we [defined an association on the model](https://github.com/18F/identity-idp/pull/10951/files#r1697593451) then we could do  `identity.email_address` right?

We could probably add some of this ownership validation as we write the attribute and maybe not worry about it at read time?